backend/src/test/java/com/hackner/musicompass/service/ArtistServiceTest.java

```diff
@@ -0,0 +1,85 @@
+            .discogsArtistApiService(testDiscogsArtistApiService)
+            .discogsSecret(testDiscogsSecret).build();
+
+    @Test
+    @DisplayName("TestArtistServiceForExistingArtist")
+    public void getArtistFromArtistService(){
+        //GIVEN
+        String artistName = "hans hammer";
+        String discogsArtistId = "6666";
+        String artistImageUrl = "https://img.discogs.com/wcD5QViPvggOaC3T_D5ql1rHY_E=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-12596-1565996731-5432.jpeg.jpg";
+        String discogsArtistUrl = "https://api.discogs.com/artists/6666";
+        String testAccessToken = "testtest";
+
+        Artist testArtist = Artist.builder()
+                .artistName(artistName)
+                .artistImageUrl(artistImageUrl)
+                .discogsId(discogsArtistId)
+                .discogsArtistUrl(discogsArtistUrl).build();
+
+        DiscogsArtist testDiscogsArtist = DiscogsArtist.builder()
+                .discogsArtistId(discogsArtistId)
+                .artistName(artistName)
+                .artistImageUrl(artistImageUrl)
+                .discogsArtistUrl(discogsArtistUrl).build();
+
+        DiscogsArtistSearchResults testDiscogsArtistSearchResults = DiscogsArtistSearchResults.builder()
+                .results(new DiscogsArtist[]{testDiscogsArtist}).build();
+
+        when(testDiscogsSecret.getDiscogsToken()).thenReturn(testAccessToken);
+        when(testDiscogsArtistApiService.getDiscogsArtistByArtistName(artistName, testDiscogsSecret.getDiscogsToken())).thenReturn(testDiscogsArtistSearchResults);
+
+        //WHEN
+        Optional<Artist> actual = testArtistService.getArtistByArtistName(artistName);
+
+        //THEN
+        assertThat(actual.get(), equalTo(testArtist));
+    }
+
+    @Test
+    @DisplayName("TestArtistServiceForNonExistingArtist")
+    public void getExceptionFromArtistService(){
+        //GIVEN
+        String artistName = "bliblablubb";
+        String discogsArtistId = "6666";
+        String artistImageUrl = "https://img.discogs.com/wcD5QViPvggOaC3T_D5ql1rHY_E=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-12596-1565996731-5432.jpeg.jpg";
+        String discogsArtistUrl = "https://api.discogs.com/artists/6666";
+        String testAccessToken = "testtest";
+
+        DiscogsArtistSearchResults testDiscogsArtistSearchResults = DiscogsArtistSearchResults.builder()
+                .results(new DiscogsArtist[]{}).build();
+
+        when(testDiscogsSecret.getDiscogsToken()).thenReturn(testAccessToken);
+        when(testDiscogsArtistApiService.getDiscogsArtistByArtistName(artistName, testDiscogsSecret.getDiscogsToken()))
+                .thenReturn(testDiscogsArtistSearchResults);
+
+        //WHEN
+        Optional<Artist> actual = testArtistService.getArtistByArtistName(artistName);
+
+        //THEN
+        assertThat(actual, Matchers.is(Optional.empty()));
```

Comment:
for better readability you could make `Matchers.is` a static import

Reply:
thx. done.